src/app/core/guards/auth/auth.guard.ts

```diff
@@ -37,7 +37,8 @@ class AuthGuard implements CanActivate, CanLoad, OnDestroy {
   }
 
   ngOnDestroy(): void {
-    this.subscription.unsubscribe();
+    this.destroy$.next(true);
+    this.destroy$.unsubscribe();
```

Comment:
is it? or it should be:

```suggestion
    this.destroy$.next();
    this.destroy$.complete();
```